File: src/BibliotecaCorporativa/backend/BibCorp.Domain/Models/Emprestimos/Emprestimo.cs
using BibCorp.Domain.Models.Acervos;
using BibCorp.Domain.Models.Patrimonios;

namespace BibCorp.Domain.Models.Emprestimos
{
  public class Emprestimo
  {
    public int Id { get; set; }
    public string UserName { get; set; }
    public TipoStatusEmprestimo Status { get; set; }
    public DateTime DataEmprestimo { get; set; }
    public DateTime DataPrevistaDevolucao { get; set; }
    public int QtdeDiasEmprestimo { get; set; }
    public DateTime? DataDevolucao { get; set; }
    public int QtdeDiasAtraso { get; set; }
    public int AcervoId { get; set; }
    public Acervo Acervo { get; set; }
    public int PatrimonioId { get; set; }
    public Patrimonio Patrimonio { get; set; }
    public string LocalDeColeta { get; set; }
    public string LocalDeEntrega { get; set; }
  }
}

File: src/BibliotecaCorporativa/frontend/BibCorp-app/src/app/emprestimos/models/emprestimo/Emprestimo.ts
import { Acervo } from "src/app/acervos";
import { Patrimonio } from "src/app/patrimonios";

export class Emprestimo {
  id: number;
  userName: string;
  status: Status;
  dataEmprestimo: Date;
  dataPrevistaDevolucao: Date;
  qtdeDiasEmprestimo: number;
  dataDevolucao: Date;
  qtdeDiasAtraso: number;
  acervoId: number;
  acervo: Acervo;
  patrimonioId: number;
  patrimonio: Patrimonio;
  localDeColeta: string;
  localDeEntrega: string;
}

export enum Status {
  Reservado = 1,
  Emprestado = 2,
  Devolvido = 3,
  Renovado = 4,
  Recusado = 5,
}
